lib/ui/chip_bar.dart

```diff
@@ -20,7 +20,7 @@ class _ChipBarState extends State<ChipBar> {
       chipLabels.forEach((k, value) {
         if (k == key) {
           chipLabels[k] = !value;
-          context.read<UserProvider>().toggleFilter(key);
+          context.read<FiltersNotifier>().toggleFilter(key);
```

Comment:
Let's make ChipBar a StatelessWidget. Make chipLabels a list and NOT a map. We won't be managing the state here - we will do it in FiltersNotifier. You already did a good job with that notifier. Use it to manage the UI states of chips